### PR TITLE
separate component

### PR DESCRIPTION
**Pull request by mydmdm**
Since the protocol allow different version between job (root) and components (e.g. docker image, data in `prerequisites`), thus it is better to separate them to independent components

### CODE REVIEW

File: schemas/v2/schema.yaml [hunk 1]
```diff
@@ -43,403 +36,413 @@ components:
...
-                    uri:
-                      type: array
-                      items:
-                        type: string
-                      minItems: 1
-                  required:
-                    - type
-                    - uri
-          minItems: 1
+    DockerImageComponent:
```
Comment (abuccts) [inline schemas/v2/schema.yaml]: could you rename to DockerImagePrerequisite?

File: schemas/v2/schema.yaml [hunk 2]
```diff
@@ -10,9 +10,8 @@ externalDocs:
 paths: {}
 components:
   schemas:
-    JobProtocol:
+    OpenPAIComponent:
```
Comment (abuccts) [inline schemas/v2/schema.yaml]: maybe rename to PrerequisiteInfo?

File: schemas/v2/schema.yaml [hunk 3]
```diff
@@ -43,403 +36,413 @@ components:
...
+                uri: hdfs://10.151.40.179:9000/core/cifar10_model
+            script_example:
+              value:
+                protocolVersion: 2
+                name: tensorflow_cnnbenchmarks
+                type: script
+                version: 84820935288cab696c9c2ac409cbd46a1f24723d
+                contributor: MaggieQi
+                description: tensorflow benchmarks
+                uri: github.com/MaggieQi/benchmarks
```
Comment (abuccts) [inline schemas/v2/schema.yaml]: the examples are not correct, pls use the original array example in `prerequisites` instead

File: schemas/v2/schema.yaml [hunk 4]
```diff
@@ -43,403 +36,413 @@ components:
         description:
           type: string
           example: Here is the job description.
```
Comment (abuccts) [inline schemas/v2/schema.yaml]: remove "job", if it's made independent

File: schemas/v2/schema.yaml (hunk 1 above)
Comment (mydmdm) [reply schemas/v2/schema.yaml]: Done. the term "Component" has been replaced by "Prerequisite"

File: schemas/v2/schema.yaml (hunk 2 above)
Comment (mydmdm) [reply schemas/v2/schema.yaml]: Done

File: schemas/v2/schema.yaml (hunk 3 above)
Comment (mydmdm) [reply schemas/v2/schema.yaml]: move the Prerequisites example back to `prerequisites` in `JobProtocol` other than in separate Prerequisite

File: schemas/v2/schema.yaml (hunk 4 above)
Comment (mydmdm) [reply schemas/v2/schema.yaml]: Done

**Review by abuccts — APPROVED** [review]

**Pull request merged**